spot-client/src/spot-remote/app-state/actions.js

```diff
@@ -22,41 +70,167 @@ import {
  */
 export function connectToSpotTV(joinCode, shareMode) {
     return (dispatch, getState) => {
+        if (remoteControlClient.getConnectPromise()) {
+            throw new Error('Spot Remote XMPP connection is still alive');
+        }
+
         dispatch({
             type: SPOT_REMOTE_WILL_VALIDATE_JOIN_CODE,
             joinCode,
             shareMode
         });
 
+        _setSubscriptions({
+            dispatch,
+            getState
+        });
+
+        const state = getState();
         const { joinCodeServiceUrl } = getSpotServicesConfig(getState());
 
         return remoteControlClient.exchangeCode(joinCode, { joinCodeServiceUrl })
-            .then(roomInfo => {
+            .then(roomInfo => remoteControlClient.connect({
+                autoReconnect: true,
```

Comment:
Is autoReconnect ever false? It might have existed as a temporary variable while moving reconnect logic into RCS. Probably a future cleanup you or I should do.

Reply:
true

Reply:
added as a separate commit